game/requesthandler.cc

```diff
@@ -1,376 +1,235 @@
+			return;
+		}
+		const int newPosition = jsonRoot["position"];
+		if(newPosition < 0) {
+			res.status = 404;
+			res.set_content("cannot find the selected position", "text/plain");
+			return;
+		}
+		Game* gm = Game::getSingletonPtr();
+		if(gm->getCurrentPlayList().isEmpty()) {
+			res.status = 404;
+			res.set_content("playlist is empty", "text/plain");
+			return;
+		}
+		const int playlistSize = gm->getCurrentPlayList().getList().size();
+		if(newPosition > playlistSize - 1) {
+			res.status = 404;
+			res.set_content("new position is out of bound", "text/plain");
+			return;
+		}
+		gm->getCurrentPlayList().setPosition(songId, newPosition);
                 ScreenPlaylist* m_pp = dynamic_cast<ScreenPlaylist*>(gm->getScreen("Playlist"));
                 m_pp->triggerSongListUpdate();
-                request.reply(web::http::status_codes::OK, "success");
-                return;
-            } else  {
-                request.reply(web::http::status_codes::BadRequest, "Not gonna move the song to \""+ std::to_string(positionToMoveTo + 1) + "\" since the list ain't that long. Please make a valid request.");
-                return;
-            }
-        } catch(web::json::json_exception const & e) {
-            std::string str = std::string("JSON Exception: ") + e.what();
-            request.reply(web::http::status_codes::BadRequest, str);
-            return;
-        }
-    } else if(path == "/api/search") {
-        auto query = jsonPostBody["query"].as_string();
-        m_songs.setFilter(query);
-        web::json::value jsonRoot = web::json::value::array();
-        for(int i = 0; i < m_songs.size(); i++) {
-            web::json::value songObject = web::json::value::object();
-            songObject["Title"] = web::json::value::string(m_songs[i]->title);
-            songObject["Artist"] = web::json::value::string(m_songs[i]->artist);
-            songObject["Edition"] = web::json::value::string(m_songs[i]->edition);
-            songObject["Language"] = web::json::value::string(m_songs[i]->language);
-            songObject["Creator"] = web::json::value::string(m_songs[i]->creator);
-            jsonRoot[i] = songObject;
-        }
-        request.reply(web::http::status_codes::OK, jsonRoot);
-        return;
-    } else {
-        request.reply(web::http::status_codes::NotFound, "The path \""+ path +"\" was not found.");
-        return;
-    }
+		res.set_content("success", "text/plain");
+	});
+	m_server.Post("/api/search", [&](const httplib::Request &, httplib::Response& res, const httplib::ContentReader &contentReader) {
+		std::string buffer;
+		contentReader([&](const char *data, size_t dataLength) {
+			buffer.append(data, dataLength);
+			return true;
+		});
+		const auto jsonRootReq = nlohmann::json::parse(buffer);
+		const std::string query = jsonRootReq["query"];
+		m_songs.setFilter(query);
+		auto jsonRoot = nlohmann::json::array();
+		for (size_t i=0; i< m_songs.size(); i++) {
```

Comment:
can't we use range-based loop here?

Reply:
We can if we implement custom iterators for the class, as I did on my PR.